packages/component/src/Middleware/ActivityGrouping/ui/PartGrouping/private/PartGroupingActivity.tsx

```diff
@@ -0,0 +1,212 @@
+import { reactNode } from '@msinternal/botframework-webchat-react-valibot';
+import { useStyles } from '@msinternal/botframework-webchat-styles/react';
+import { hooks } from 'botframework-webchat-api';
+import { getOrgSchemaMessage, type WebChatActivity } from 'botframework-webchat-core';
+import React, { memo, useCallback, useMemo, useState, type MouseEventHandler } from 'react';
+import {
+  any,
+  array,
+  minLength,
+  object,
+  optional,
+  parse,
+  pipe,
+  readonly,
+  string,
+  transform,
+  type InferOutput
+} from 'valibot';
+
+import StackedLayoutMain from '../../../../../Activity/StackedLayoutMain';
+import StackedLayoutRoot from '../../../../../Activity/StackedLayoutRoot';
+import StackedLayoutStatus from '../../../../../Activity/StackedLayoutStatus';
+import useActivityElementMapRef from '../../../../../providers/ChatHistoryDOM/useActivityElementRef';
+import useActiveDescendantId from '../../../../../providers/TranscriptFocus/useActiveDescendantId';
+import useGetGroupDescendantIdByActivityKey from '../../../../../providers/TranscriptFocus/useGetGroupDescendantIdByActivityKey';
+import useFocusByGroupKey from '../../../../../providers/TranscriptFocus/useFocusByGroupKey';
+import FocusTrap from '../../../../../Transcript/FocusTrap';
+import useRenderActivityProps from '../../../../../Transcript/hooks/useRenderActivityProps';
+import {
+  TranscriptActivityList,
+  TranscriptFocusContent,
+  TranscriptFocusContentOverlay,
+  TranscriptFocusContentActiveDescendant,
+  TranscriptFocusIndicator
+} from '../../../../../Transcript/TranscriptFocus';
+import isZeroOrPositive from '../../../../../Utils/isZeroOrPositive';
+import { android } from '../../../../../Utils/detectBrowser';
+import CollapsibleGrouping from '../CollapsibleGrouping';
+import usePartGroupingLogicalGroup from './usePartGroupingLogicalGroup';
+
+import styles from './PartGroupingActivity.module.css';
+
+const { useAvatarForBot, useStyleOptions, useGetKeyByActivity } = hooks;
+
+const partGroupingActivityPropsSchema = pipe(
+  object({
+    activities: pipe(
+      array(
+        pipe(
+          any(),
+          transform(value => value as WebChatActivity)
+        )
+      ),
+      minLength(1, 'botframework-webchat: "activities" must have at least 1 activity'),
+      readonly()
+    ),
+    children: optional(reactNode())
+  }),
+  readonly()
+);
+
+type PartGroupingActivityProps = InferOutput<typeof partGroupingActivityPropsSchema>;
+
+const partGroupingFocusableActivityPropsSchema = pipe(
+  object({
+    activity: pipe(
```

Comment:
This will call `Object.freeze()` on the activity.

IMO, the input activity should be frozen, but we are not there yet.